Added the Webhook support.

deniskristov opened the pull request.

deniskristov committed:
Added the Webhook support.

roborox commented:
Привет @deniskristov 
спасибо за PR! Работа через webhook'и - это полезная вещь, только что узнал, что так вообше можно )

Есть несколько вещей, которые хорошо бы поправить, чтобы я мог принять PR

1. форматирование кода. Открывающаяся фигурная скобка должна стоять не на новой строке, а на той же
2. зависимость lombok не должна быть compile. Provided. https://projectlombok.org/mavenrepo/
3. лучше перевести все сущности на lombok, если добавляете такую зависимость
4. в новом коде есть неиспользуемые поля и переменные. проверьте, пожалуйста

Возможно, будут еще замечания. после исправления посмотрю еще раз
Спасибо

roborox commented:
@deniskristov а где описание запросов с помощью json в body? что-то не нашел в доке битрикса

deniskristov committed:
Added the Webhook support.
- corrected the code format
- all entities should use lombok annotations
- changed @Test annotation to 'org.testing project' to run tests during a maven lifecycle
- removed unused fields

deniskristov commented:
Привет @roborox 
Все пожелания учел в новом комите, еще поправил аннотации @TEST в тестах, вместо JUNIT использовал org.testng, иначе тесты не запускались в момент выполнения maven lifecycle, так как именно org.testng подключен в pom.xml

deniskristov commented:
Описания JSON запросов смотрел тут https://dev.1c-bitrix.ru/rest_help/crm/index.php
Например создание новой сделки:
```
{
        fields:
        { 
            "TITLE": "Плановая продажа", 
            "TYPE_ID": "GOODS", 
            "STAGE_ID": "NEW", 					
            "COMPANY_ID": 3,
            "CONTACT_ID": 3,
            "OPENED": "Y", 
            "ASSIGNED_BY_ID": 1, 
            "PROBABILITY": 30,
            "CURRENCY_ID": "USD", 
            "OPPORTUNITY": 5000,
            "BEGINDATE": date2str(current),
            "CLOSEDATE": date2str(nextMonth)					
        },
        params: { "REGISTER_SONET_EVENT": "Y" }	
}
```

roborox commented:
@deniskristov я имел ввиду для фильтров, а не для создания/изменения сущностей

deniskristov commented:
все там же, вот, например, поиск контактов по фильтру:
https://dev.1c-bitrix.ru/rest_help/crm/contacts/crm_contact_list.php
я проверил свой код на рабочей системе, поиск работет

deniskristov commented:
Вот рабочий пример поиска контакта по телефону:
```
String phone = "123456789";
FilterBuilder builder = new FilterBuilder();
builder.put("PHONE", phone);
BitrixPage<BitrixContact> bitrixPage = bitrixApi.listContacts(null, builder.build());
```

roborox commented:
@deniskristov смержил
также упростил немного. в Webhook client не нужны tokens. Перенес их только в старый ClientImpl
